internal/check/engine.go

```diff
@@ -90,6 +111,11 @@ func (e *Engine) checkOneIndirectionFurther(ctx context.Context, requested *rela
 	}
 }
 
-func (e *Engine) SubjectIsAllowed(ctx context.Context, r *relationtuple.InternalRelationTuple) (bool, error) {
-	return e.checkOneIndirectionFurther(ctx, r, &relationtuple.RelationQuery{Object: r.Object, Relation: r.Relation, Namespace: r.Namespace})
+func (e *Engine) SubjectIsAllowed(ctx context.Context, r *relationtuple.InternalRelationTuple, restDepth int) (bool, error) {
+	// global max-depth takes precedence when it is the lesser or if the request max-depth is less than or equal to 0
+	if globalMaxDepth := e.d.Config().ReadAPIMaxDepth(); restDepth <= 0 || globalMaxDepth < restDepth {
```

Comment:
I see, so `0` means fallback to global.

Reply:
Yeah, I tried to follow the logic I found in the codebase. Instead of validation and errors there are safe defaults or sane response. Maybe some follow up work could be more explicit in validating and providing detailed feedback to the user

Reply:
Sounds good to me :+1: